sparkler-app/src/main/scala/edu/usc/irds/sparkler/solr/SolrUpsert.scala

```diff
@@ -38,9 +37,16 @@ class SolrUpsert(job: SparklerJob) extends ((TaskContext, Iterator[Resource]) =>
     val solrClient = job.newCrawlDbSolrClient()
 
     //TODO: handle this in server side - tell solr to skip docs if they already exist
-    val newResources: Iterator[Resource] = for (doc <- docs if solrClient.crawlDb.getById(doc.getId) == null) yield doc
+    var newResources: Set[Resource] = Set()
+    for (doc <- docs) {
+      val qry: SolrQuery = new SolrQuery(Constants.solr.STATUS + ":" + ResourceStatus.UNFETCHED)
+      qry.addFilterQuery(s"${Constants.solr.DEDUPE_ID}:${doc.getDedupeId}")
```

Comment:
@karanjeets 
So, this query looks for `status:UNFETCHED AND dedub_id:<dedupe_id>`

My question is why `status:UNFETCHED`  filter required? Do you mean to overwrite already fetched?

Is it a bug or am I missing something?